components/GameScreen.vue

```diff
@@ -179,87 +260,55 @@ watch(gameState, async (newState, prevState) => {
       autoDensity: true,
     });
 
-    let isZoomingOut = false;
-    const zoomDuration = 500; // Duration of the zoom-out effect in milliseconds
-    let startZoomTime: number | null = null;
-    let startMousePos = { x: 0, y: 0 };
-
-    // Functions to handle smooth zoom out
-    function smoothZoomOut(mousePos: { x: number; y: number }) {
-      if (!appRef.value) {
+    // Follow player bot
+    function follow() {
+      isFollowing = !isFollowing;
+      const followButton = document.getElementById("followButton");
+      if (followButton) {
+        followButton.textContent = isFollowing ? "stop following my bot" : "follow my bot";
+      }
+      if (!gameState.value) {
         return;
       }
 
-      isZoomingOut = true;
-      startZoomTime = performance.now();
-      startMousePos = mousePos;
-      requestAnimationFrame(animateZoomOut);
+      let playerBot = null;
+      for (const bot of Object.values(gameState.value.bots)) {
+        if (bot.username == user.value?.body.username) {
+          playerBot = bot;
+        }
+      }
+      // const firstBot = Object.values(gameState.value.bots)[1];
+      if (playerBot) {
+        const newScale = Math.max(minZoom, Math.min(maxZoom, app.stage.scale.x + 10));
```

Comment:
Should we take into account bot size when computing new scale so that larger bots still fit into the zoomed in viewport?

Reply:
I'm not sure what this means.
Cause the max zoom in level should be large enough to fit the larger bot in the viewport.

If theirs a case that the bot got too large for the viewport, I think its better remains the same. To preserve their intensity of the largeness of the bot that got too large.

Reply:
@aldrin312, the largest bot can have a radius of 150px (diameter of 300px).

> To preserve their intensity of the largeness of the bot that got too large.

If it doesn't fit into the zoomed-in view, it will make the view itself useless and will look and feel like a bug.

If there is a straightforward way to account for the bot size when zooming (zoom a bit less for larger bots), it would be great to add it to this PR. Otherwise, we can see how often this is a problem.

Reply:
Oh unless you mean the size of the bot that is being followed.

Reply:
@aldrin312, exactly 😄